### PR TITLE
sistema de musica

### PR DESCRIPTION
**Pull request by NapoElWapo**

### CODE REVIEW

File: missile_toad/include/missile_toad/core/music.hpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        Music(raylib::Music &&music, std::unique_ptr<uint8_t[]> &&buffer) noexcept;
+        Music &operator=(Music &&) noexcept = default;
+
+        /**
+         * Get the underlying raylib music.
+         * @return The underlying raylib music.
+         */
+        [[nodiscard]] const raylib::Music &get_music() const noexcept;
+
+        [[nodiscard]] raylib::Music &get_music() noexcept;
```
Comment (Alan5142) [inline missile_toad/include/missile_toad/core/music.hpp:37]: Add documentation

File: missile_toad/main.cpp [hunk 2]
```diff
@@ -64,7 +64,7 @@ int main(int argc, char *argv[]) noexcept(false)
 #ifdef PLATFORM_NX
     romfsInit();
 #endif
-
+    InitAudioDevice();
```
Comment (Alan5142) [inline missile_toad/main.cpp:68]: Consider adding a newline before and after this line,

File: missile_toad/main.cpp [hunk 3]
```diff
@@ -101,7 +101,7 @@ int main(int argc, char *argv[]) noexcept(false)
     }
 
     spdlog::info("Game ended.");
-
+    CloseAudioDevice();
```
Comment (Alan5142) [inline missile_toad/main.cpp:107]: Consider adding a newline before and after this line,

**Review by Alan5142 — CHANGES_REQUESTED** [review]

File: missile_toad/include/missile_toad/core/music.hpp (hunk 1 above)
Comment (NapoElWapo) [reply missile_toad/include/missile_toad/core/music.hpp:37]: unu

**Comment by Alan5142** [comment]
@NapoElWapo add these
```yaml
-cppcoreguidelines-avoid-c-arrays
-modernize-avoid-c-arrays
```
to `.clang-tidy` file under the `Checks` section

**Comment by Alan5142** [comment]
* Create a component called AudioComponent, this component must contain a `entt::resource<missiletoad::core::Music>`. To do this, execute the command `python scripts/missiletoadcli.py component core::AudioComponent`

* Manually updating the music is tedious, that's why we should create a new System! `AudioSystem`, this system should get all components of type `AudioComponent` and update them in the `on_update` member function of the System. Add this system to the Scene class. To create this, execute the command `python scripts/missiletoadcli.py system core::AudioSystem`. See the following code to check where to create the instance.

https://github.com/Alan5142/missile-toad/blob/b92b32f15aa7534c91d1309fdbc3c4e19c565a15/missile_toad/src/core/scene.cpp#L72-L76

File: missile_toad/include/missile_toad/core/music.hpp (hunk 1 above)
Comment (Alan5142) [reply missile_toad/include/missile_toad/core/music.hpp:37]: Seems good

File: missile_toad/include/missile_toad/core/systems/audio.system.hpp
Comment (Alan5142) [inline missile_toad/include/missile_toad/core/systems/audio.system.hpp:1]: Override on_update and call music update for all AudioComponents, that's the change I requested for AudioSystem.
Check entt::registry to get all components of certain type.

**Review by Alan5142 — CHANGES_REQUESTED** [review]
Call update for each music component in the music system.

File: missile_toad/include/missile_toad/core/systems/audio.system.hpp
Comment (NapoElWapo) [reply missile_toad/include/missile_toad/core/systems/audio.system.hpp:1]: why do you hate me so much

Comment (Alan5142) [reply missile_toad/include/missile_toad/core/systems/audio.system.hpp:1]: Change is not ready tho

Comment (NapoElWapo) [reply missile_toad/include/missile_toad/core/systems/audio.system.hpp:1]: i'm, confused unu

Comment (Alan5142) [inline missile_toad/include/missile_toad/core/systems/audio.system.hpp:1]: If your destructor is empty, you could just `default`ed it.

I'm not going to get picky with it tho

File: missile_toad/src/core/systems/audio.system.cpp
Comment (Alan5142) [inline missile_toad/src/core/systems/audio.system.cpp:1]: Same as the other one, `default`ed would be better but not necessary

**Review by Alan5142 — COMMENTED** [review]
Fix clang-tidy issues and we're done

**Review by Alan5142 — APPROVED** [review]
Ok, just wait for the checks

**Pull request merged**